sklearn/utils/tests/test_array_api.py

```diff
@@ -581,10 +581,14 @@ def test_count_nonzero(
 @pytest.mark.parametrize("wrap", [True, False])
 def test_fill_or_add_to_diagonal(array_namespace, device_, dtype_name, wrap):
     xp = _array_api_for_tests(array_namespace, device_)
-    array_np = numpy.zeros((5, 4), dtype=numpy.int64)
-    array_xp = xp.asarray(array_np)
-    _fill_or_add_to_diagonal(array_xp, value=1, xp=xp, add_value=False, wrap=wrap)
+
+    array_np = numpy.zeros((5, 4), dtype=dtype_name)
+    array_xp = xp.asarray(array_np, device=device_)
```

Comment:
Maybe we could make it explicit that this test does not rely on possible shared memory across namespaces when the namespace/device pair allows for it.

```suggestion
    array_xp = xp.asarray(array_np.copy(), device=device_)
```